src/main/java/org/folio/bulkops/service/RecordUpdateService.java

```diff
@@ -0,0 +1,35 @@
+package org.folio.bulkops.service;
+
+import static org.folio.bulkops.util.Utils.resolveEntityClass;
+
+import lombok.RequiredArgsConstructor;
+import org.folio.bulkops.domain.bean.BulkOperationsEntity;
+import org.folio.bulkops.domain.bean.StateType;
+import org.folio.bulkops.domain.entity.BulkOperation;
+import org.folio.bulkops.domain.entity.BulkOperationExecutionContent;
+import org.folio.bulkops.processor.UpdateProcessorFactory;
+import org.folio.bulkops.repository.BulkOperationExecutionContentRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class RecordUpdateService {
+  private final UpdateProcessorFactory updateProcessorFactory;
+  private final BulkOperationExecutionContentRepository executionContentRepository;
+
+  public BulkOperationsEntity updateEntity(BulkOperationsEntity original, BulkOperationsEntity modified, BulkOperation operation) {
+    var isEqual = original.hashCode() == modified.hashCode() && original.equals(modified);
```

Comment:
I'm not sure, the hashCode values equality check is needed:
hashCode() and equals() contract states that if two objects are equal to each other based on equals() method, then the hash code must be the same